### PR TITLE
[DOCS] Added compatibilty doc + updated README

### PR DESCRIPTION
**Pull request by ezimuel**
THis PR adds a compatibility section + updates the README.

### CODE REVIEW

File: docs/compatibility.asciidoc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+== Compatibility
+
+Language clients are forward compatible; meaning that clients support communicating
+with greater or equal minor versions of Elasticsearch. Elasticsearch language
+clients are only backwards compatible with default distributions and without
+guarantees made.
```
Comment (szabosteve) [inline docs/compatibility.asciidoc:6]: In the rest of the Client books, we added the compatibility information to the `Installation` section as a level-three sub-section:
```
[discrete]
[[compatibility]]
=== Compatibility
```
If there is no particular reason to add this info as a separate section, I would suggest moving this content under the Installation section.

File: docs/index.asciidoc [hunk 2]
```diff
@@ -8,6 +8,6 @@ include::overview.asciidoc[]
 
 include::installation.asciidoc[]
 
-
+include::compatibility.asciidoc[]
```
Comment (szabosteve) [inline docs/index.asciidoc:11]: If you move the compatibility info under `Installation`, then this item can be deleted.
```suggestion
```

**Review by szabosteve — COMMENTED** [review]
One suggestion about the place of the compatibility info inside the Perl book.

**Review by ezimuel — COMMENTED** [review]
I just moved the compatibility section into installation

**Review by szabosteve — APPROVED** [review]
LGTM!

**Pull request merged**